src/components/elements/annotationSidePanel/components/annotationWizard/steps/AnnotationFormSegment.tsx

```diff
@@ -0,0 +1,194 @@
+                                    </p>
+
+                                    <Field name={`annotationValues.${fieldName}`}
+                                        value={fieldValue ?? ''}
+                                        className="w-100 b-grey br-corner mt-1 py-1 px-2"
+                                    />
+                                </Col>
+                            </Row>
+                        );
+                    })}
+                </div>
+            );
+        } case 'array': {
+            /* Format field name for field array sub class */
+            const fieldName = `${annotationFormFieldProperty.jsonPath.replaceAll('.', '_').replaceAll('][', '_').replaceAll('[', '').replaceAll(']', '')}`;
+
+            return (
+                <div>
+                    <FieldArray name={`annotationValues.${fieldName}`}>
+                        {({ push, remove }) => (
+                            <>
+                                <div className="bgc-primary tc-white b-grey br-corner mb-3 px-3 py-2">
+                                    <Row className="py-1">
+                                        <Col className="d-flex align-items-center">
+                                            <p className="fw-bold">
+                                                {`${annotationFormFieldProperty.name}s`}
+                                            </p>
+                                        </Col>
+                                        <Col lg="auto"
+                                            className="d-flex align-items-center"
+                                        >
+                                            <Button type="button"
+                                                variant="blank"
+                                                className="px-0 py-0"
+                                                OnClick={() => push({})}
+                                            >
+                                                <FontAwesomeIcon icon={faPlus}
+                                                    size="lg"
+                                                />
+                                            </Button>
+                                        </Col>
+                                    </Row>
+                                </div>
+
+                                {/* For each sub class index, render an annotation form segment */}
+                                {formValues?.annotationValues?.[fieldName]?.map((_classObject: Dict, index: number) => {
+                                    /* Set sub class as object and render form segment */
+                                    const key = `${fieldName}_${index}`;
+                                    const annotationFormFieldSubProperty: AnnotationFormProperty = { ...annotationFormFieldProperty };
+
+                                    annotationFormFieldSubProperty.type = 'object';
```

Comment:
Can you set this in the previous line to avoid mutating this object

```
const annotationFormFieldSubProperty: AnnotationFormProperty = { ...annotationFormFieldProperty, type: 'object' };

```
?

Reply:
Yes did that